compiler-rt/lib/builtins/int_types.h

```diff
@@ -189,8 +189,10 @@ typedef long double tf_float;
 #define CRT_LDBL_IEEE_F128
 #endif
 #define TF_C(x) x##L
-#elif __LDBL_MANT_DIG__ == 113
-// Use long double instead of __float128 if it matches the IEEE 128-bit format.
+#elif __LDBL_MANT_DIG__ == 113 ||                                              \
+    (__FLT_RADIX__ == 16 && __LDBL_MANT_DIG__ == 28)
+// Use long double instead of __float128 if it matches the IEEE 128-bit format
+// or the IBM hexadecimal format.
 #define CRT_LDBL_128BIT
 #define CRT_HAS_F128
 #define CRT_HAS_IEEE_TF
```

Comment:
These last two macros should only be defined for the `__LDBL_MANT_DIG__ == 113` case.

Reply:
The CRT_HAS_F128 needs to be defined for hex float too.  The Qcomplex and COMPLEXTF_REAL/COMPLEXTF_IMAGINARY ids are only defined if CRT_HAS_F128 is defined and multc3.c/divtc3.c depend on those.

I agree with CRT_HAS_IEEE_TF.  This should also only be defined if CRT_HAS_TF_MODE is defined.

Reply:
I double checked and Qcomplex is defined via CRT_LDBL_128BIT.  I can limit CRT_HAS_F128  to IEEE too.